### DISCUSSION TITLE
pagimageview在不play的情况下进行缩放动画会消失

### DISCUSSION THREAD

**Discussion by RainlyTree** [Tencent/libpag#1682] [Q&A] [answered]
```java
void fireImageView(int id, String path) {
        PAGImageView view = findViewById(id);
        view.setPath(path);
        view.setRepeatCount(-1);
        ScaleAnimation animScaleOut = new ScaleAnimation(1f, 0.606060f, 1f, 0.606060f, ScaleAnimation.RELATIVE_TO_SELF, 0.5f, ScaleAnimation.RELATIVE_TO_SELF, 0f);
        animScaleOut.setDuration(100);
        animScaleOut.setFillAfter(true);
        view.startAnimation(animScaleOut);
    }
```

**Comment by kevingpqi123** [comment]
为了提高对接效率，请提供完整的 demo 工程，上面显示都是缺失的，View 都没有添加到容器中，我们没有太多时间帮大家构造复现工程
如果你不太清楚如何上传附件，将 demo 工程上传到自己的 github 仓库中，提供 github 仓库链接也可以

**Reply by RainlyTree** [reply]
https://github.com/RainlyTree/Android-studio/tree/master/pag-android

**Reply by RainlyTree** [reply]
官方提供的demo 倒数第二行的pagimageview进行了修改

**Reply by kevingpqi123** [reply]
1、后续反馈问题的时候建议不要仅仅丢一个工程出来，更重要的是要告诉我们你修改了什么地方，比如这里截个图就可以，是最清晰的
<img width="2065" alt="企业微信截图_54d57915-04cf-455b-9abf-47e3f550636b" src="https://github.com/Tencent/libpag/assets/16609221/3b24a1ce-7099-4fbb-b050-b8eb403c3e73">

2、这里的缩放动画不是消失了，而是缩放动画太短，只有 100ms，转瞬即逝，你设置成 5000ms 看下，或者添加一个 click 的操作，当你看到时再点击触发

**Reply by RainlyTree** [reply]
https://github.com/Tencent/libpag/assets/49740526/d020598e-26b9-4b64-93d8-1fa6d000dc42
看一下这个视频 不是动画消失 而是pag内容不显示了

**Reply by kevingpqi123** [reply]
我这边使用 4.3.18 在 小米上测试是正常的，另外如果需要显示当前帧数据，正确的使用方法是需要调用 flush 方法， 由于历史原因，之前setPath 内置了这个操作

**Reply by RainlyTree** [reply]
我这里使用的也是4.3.18 oppo R11手机 小米note 2都会出现这个情况 并且在动画未开始/结束时调用flush方法均无效

**Comment by h3r3x3** [comment, marked as answer]
这里在小米note3上看了下，在第一次调用onSizeChange以后，画面其实已经刷新出来了。但是这些手机用一个有点差异的size再次触发一次onSizeChange，这个函数内部会release调之前的bitmap，如果没有被再次触发flush，是不会重新刷新的。
这里可以考虑调用flush的时候加一个延迟，比如200ms这样来规避。 
后续的话我们看下这里要不要加一个强制的后台刷新逻辑。
